diff -Naur -x '*.rej' -x '*.orig' -x build -x awk_x86_64_defconfig linux-5.5_stock/drivers/gpu/drm/drm_vblank.c linux-5.5_15kHz/drivers/gpu/drm/drm_vblank.c
--- linux-5.5_stock/drivers/gpu/drm/drm_vblank.c	2020-01-27 01:23:03.000000000 +0100
+++ linux-5.5_15kHz/drivers/gpu/drm/drm_vblank.c	2020-02-01 17:50:11.824009410 +0100
@@ -276,8 +276,7 @@
 		      cur_vblank, vblank->last);
 
 	if (diff == 0) {
-		WARN_ON_ONCE(cur_vblank != vblank->last);
-		return;
+                diff = 1;
 	}
 
 	/*
