diff -Naur -x '*.rej' -x '*.orig' -x build -x awk_x86_64_defconfig linux-5.5_stock/drivers/gpu/drm/drm_modes_switchres.c linux-5.5_15kHz/drivers/gpu/drm/drm_modes_switchres.c
--- linux-5.5_stock/drivers/gpu/drm/drm_modes_switchres.c	2020-02-01 17:53:14.287334239 +0100
+++ linux-5.5_15kHz/drivers/gpu/drm/drm_modes_switchres.c	2020-02-01 20:13:48.297106161 +0100
@@ -32,6 +32,10 @@
 	{ DRM_MODE("320x240", DRM_MODE_TYPE_DRIVER, 6640, 320, 336,
 			368, 424, 0, 240, 242, 245, 261, 0,
 			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 640x240@60.00 15 Khz */
+	{ DRM_MODE("640x240", DRM_MODE_TYPE_DRIVER, 13220, 640, 672,
+			736, 832, 0, 240, 243, 246, 265, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
 	/* 640x480@60.00 15.750 Khz */
 	{ DRM_MODE("640x480i", DRM_MODE_TYPE_DRIVER, 13104, 640, 664,
 			728, 832, 0, 480, 484, 490, 525, 0,
