diff -Naur linux-5.3_stock/drivers/gpu/drm/drm_vblank.c linux-5.3/drivers/gpu/drm/drm_vblank.c
--- linux-5.3_stock/drivers/gpu/drm/drm_vblank.c	2019-09-15 23:19:32.000000000 +0200
+++ linux-5.3/drivers/gpu/drm/drm_vblank.c	2019-09-16 11:14:16.545141473 +0200
@@ -276,8 +276,7 @@
 		      pipe, vblank->count, diff, cur_vblank, vblank->last);
 
 	if (diff == 0) {
-		WARN_ON_ONCE(cur_vblank != vblank->last);
-		return;
+                diff = 1;
 	}
 
 	/*
