diff -Naur linux-5.0_stock/drivers/gpu/drm/drm_vblank.c linux-5.0/drivers/gpu/drm/drm_vblank.c
--- linux-5.0_stock/drivers/gpu/drm/drm_vblank.c        2019-03-03 23:21:29.000000000 +0000
+++ linux-5.0/drivers/gpu/drm/drm_vblank.c      2019-03-04 09:35:58.690014048 +0000
@@ -258,8 +258,7 @@
 		      pipe, vblank->count, diff, cur_vblank, vblank->last);
 
 	if (diff == 0) {
-		WARN_ON_ONCE(cur_vblank != vblank->last);
-		return;
+                diff = 1;
 	}
 
 	/*
