diff -Naur linux-5.17_stock/drivers/gpu/drm/drm_vblank.c linux-5.17_15kHz/drivers/gpu/drm/drm_vblank.c
--- linux-5.17_stock/drivers/gpu/drm/drm_vblank.c	1970-01-01 01:00:00.000000000 +0100
+++ linux-5.17_15kHz/drivers/gpu/drm/drm_vblank.c	1970-01-01 01:00:00.000000000 +0100
@@ -351,8 +351,10 @@
 		    diff, cur_vblank, vblank->last);
 
 	if (diff == 0) {
-		drm_WARN_ON_ONCE(dev, cur_vblank != vblank->last);
-		return;
+		// fix interlaced vblank
+		//drm_WARN_ON_ONCE(dev, cur_vblank != vblank->last);
+		//return;
+		diff = 1;
 	}
 
 	/*
