diff -Naur linux-5.9_stock/drivers/gpu/drm/drm_vblank.c linux-5.9_15kHz/drivers/gpu/drm/drm_vblank.c
--- linux-5.9_stock/drivers/gpu/drm/drm_vblank.c	2020-10-11 00:00:00.000000000 +0200
+++ linux-5.9_15kHz/drivers/gpu/drm/drm_vblank.c	2020-10-14 00:00:00.000000000 +0200
@@ -348,8 +348,10 @@
 		    diff, cur_vblank, vblank->last);
 
 	if (diff == 0) {
-		drm_WARN_ON_ONCE(dev, cur_vblank != vblank->last);
-		return;
+		// fix interlaced vblank
+		//drm_WARN_ON_ONCE(dev, cur_vblank != vblank->last);
+		//return;
+		diff = 1;
 	}
 
 	/*
